streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -972,8 +970,8 @@ public synchronized State state() {
         return state;
     }
 
-    private synchronized void setStateWhenNotInPendingShutdown(final State newState) {
-        if (state == State.PENDING_SHUTDOWN) {
+    private synchronized void setStateWhenNotShuttingDownOrDead(final State newState) {
+        if (state == State.PENDING_SHUTDOWN || state == State.DEAD) {
```

Comment:
Or in the close call of the thread:

```
public synchronized void close() {
        log.info("{} Informed thread to shut down", logPrefix);
        setState(State.PENDING_SHUTDOWN);
    }
```

We can first check if the thread is already not running before continue the logging and the setting of the state, would that be more clean?